### ISSUE TITLE
Add support for kdump

### ISSUE THREAD

**Issue by jonte** [Pelagicore/meta-pelux#16]
Debugging kernel crashes would be much easier if we had kdump support in the kernel & in systemd.

Basically, we would need to:
* Configure the kernel with kdump, kexec and debug support (see [1])
* Configure persistent bootloader settitings for the crashkernel parameter (depends on which bootloader we use)
* Create systemd unit for loading backup kernel using kexec (see [1])

Issues to consider
---------------------------

Always enabbling kdump may cause performance issues, so maybe we want to enable it only on debug images.

[1]: https://wiki.archlinux.org/index.php/Kdump

**Comment by jonte** [comment]
Poky has recipes for the kexec utility: https://git.yoctoproject.org/cgit.cgi/poky/plain/meta/recipes-kernel/kexec

**Comment by GordanM** [comment]
@jonte it would be reasonable to have a `recipes-kernel/linux/linux-yocto/debug-extras.cfg` file that enables kernel debugging and the developer should be responsible to configure it manually if needed? 

On another note, there are quite a few kernel debug utilities that would make sense to include in this image I think they are found under `Kernel Hacking` menuconfig.

**Comment by jonte** [comment]
@GordanM Yes, that config file makes sense. I think it would also make sense to be able to enable the whole kdump feature using some config flag (or per defailt in dbg images)or so though, for ease of use.

There are a few steps you need to go through to get it all working, and it would be nice if PELUX helped out with enabling it.
